lib/src/crop.dart

```diff
@@ -456,36 +454,42 @@ class _CropEditorState extends State<_CropEditor> {
                 child: GestureDetector(
                   onScaleStart: widget.interactive ? _startScale : null,
                   onScaleUpdate: widget.interactive ? _updateScale : null,
-                  child: Container(
-                    color: widget.baseColor,
-                    width: MediaQuery.of(context).size.width,
-                    height: MediaQuery.of(context).size.height,
-                    child: Stack(
-                      children: [
-                        Positioned(
-                          left: _imageRect.left,
-                          top: _imageRect.top,
-                          child: Image.memory(
-                            widget.image,
-                            width: _isFitVertically
-                                ? null
-                                : MediaQuery.of(context).size.width * _scale,
-                            height: _isFitVertically
-                                ? MediaQuery.of(context).size.height * _scale
-                                : null,
-                            fit: BoxFit.contain,
+                  child: Flex(
+                    direction: _isFitVertically ? Axis.vertical : Axis.horizontal,
+                    children: [
+                      Expanded(
+                        child: Container(
+                          foregroundDecoration: (widget.borderColor != null && widget.maskColor == null)
+                              ? BoxDecoration(
+                                  border: Border.all(color: widget.borderColor!, width: 2),
+                                )
+                              : null,
+                          color: widget.baseColor,
+                          width: MediaQuery.of(context).size.width,
+                          height: MediaQuery.of(context).size.height,
+                          child: Stack(
+                            children: [
+                              Positioned(
+                                left: _imageRect.left,
+                                top: _imageRect.top,
+                                child: Image.memory(
```

Comment:
I guess the border decoration should come here. Otherwise, as long as checking the example app, the border comes to the boundary of `Crop`, not the image.

<img src="https://user-images.githubusercontent.com/20849526/233518408-cd1feffb-8e35-48bb-90df-51db18a280a7.png" width=300>